src/model/AirLoopHVACOutdoorAirSystem.cpp

```diff
@@ -305,268 +353,231 @@ namespace model {
       return edges;
     }
 
-    std::vector<ModelObject> AirLoopHVACOutdoorAirSystem_Impl::oaComponents() const {
+    std::vector<ModelObject> AirLoopHVACOutdoorAirSystem_Impl::oaComponents(openstudio::IddObjectType type) const {
       std::vector<ModelObject> modelObjects;
+      modelObjects.reserve(10);  // Abritrary, but it wouldn't be common to have more objects than that
 
-      OptionalModelObject modelObject;
-
-      modelObject = this->outdoorAirModelObject();
+      OptionalModelObject modelObject = this->outdoorAirModelObject();
 
       while (modelObject) {
```

Comment:
This kind of walking the flow path and switching based on HVACComponent type has been done a lot, but is always kind of stinky to me. At some point along the way we introduced a graph traversing function in [findModelObjects](https://github.com/NREL/OpenStudio/blob/93b47f4b9b798dec8bf02037569398d329a9609a/src/model/Loop.cpp#L181). I wonder if that idea could be applied here for the oa system.